sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -688,6 +691,15 @@ public void validate() {
     private Instant curTimestamp;
     private Iterator<PartitionState> curBatch = Collections.emptyIterator();
 
+    private final Counter messagesConsumed = Metrics.counter("io", "messagesConsumed");
+    private final Counter bytesConsumed = Metrics.counter("io", "bytesConsumed");
+    private final Distribution backlogBytes = Metrics.distribution("io", "backlog.bytes");
+    private final Distribution backlogMessages = Metrics.distribution("io", "backlog.messages");
```

Comment:
Since the metrics types currently supported are `Counter` and `Distribution`. I used `Distribution` for backlog metrics. But since the most important thing to monitor regarding backlog is the current backlog, I believe something more along the lines of Codahale's `Gauge` is appropriate here.

@bjchambers WDYT? Should I open a ticket for a Gauge metric type?